configserver/src/test/java/com/yahoo/vespa/config/server/rpc/RpcTester.java

```diff
@@ -69,20 +67,38 @@ public class RpcTester implements AutoCloseable {
         this.temporaryFolder = temporaryFolder;
         this.applicationId = applicationId;
         this.tenantName = applicationId.tenant();
-        int port = allocatePort();
-        spec = createSpec(port);
-        configBuilder.rpcport(port)
-                .configServerDBDir(temporaryFolder.newFolder().getAbsolutePath())
-                .configDefinitionsDir(temporaryFolder.newFolder().getAbsolutePath())
-                .fileReferencesDir(temporaryFolder.newFolder().getAbsolutePath());
-        configserverConfig = new ConfigserverConfig(configBuilder);
-        rpcServer = createRpcServer(configserverConfig);
-        tenantRepository = new TestTenantRepository.Builder()
-                .withHostRegistry(hostRegistry)
-                .withConfigserverConfig(configserverConfig)
-                .build();
-        tenantRepository.addTenant(tenantName);
-        startRpcServer();
+
+        Spec tempSpec;
+        ConfigserverConfig tempConfig;
+        RpcServer tempRpcServer;
+        TenantRepository tempTenantRepository;
+
+        int iterations = 0;
+        // Need to loop because we might get a port that is in use
+        do {
+            int port = allocatePort();
+            tempSpec = createSpec(port);
+            configBuilder.rpcport(port)
+                    .configServerDBDir(temporaryFolder.newFolder().getAbsolutePath())
+                    .configDefinitionsDir(temporaryFolder.newFolder().getAbsolutePath())
+                    .fileReferencesDir(temporaryFolder.newFolder().getAbsolutePath());
+            tempConfig = new ConfigserverConfig(configBuilder);
+            tempRpcServer = createRpcServer(tempConfig);
+            tempTenantRepository = new TestTenantRepository.Builder()
+                    .withHostRegistry(hostRegistry)
+                    .withConfigserverConfig(tempConfig)
+                    .build();
+            tempTenantRepository.addTenant(tenantName);
+            startRpcServer(tempRpcServer, tempTenantRepository, tempSpec);
+            iterations++;
+            Thread.sleep(1);
+        } while (!tempRpcServer.isRunning() && iterations < 10);
```

Comment:
Perhaps assert here that the server is running?

Reply:
Yes, good point, done